pkg/sql/sem/builtins/fingerprint_builtin_test.go

```diff
@@ -0,0 +1,212 @@
+						if _, ok := ba.Requests[i].GetInner().(*roachpb.ExportRequest); ok {
+							exportResponse := ru.GetInner().(*roachpb.ExportResponse)
+							numExportResponses++
+							numSSTsInExportResponses += len(exportResponse.Files)
+						}
+					}
+					return nil
+				},
+			},
+		},
+	})
+
+	resetVars := func() {
+		mu.Lock()
+		defer mu.Unlock()
+		numExportResponses = 0
+		numSSTsInExportResponses = 0
+	}
+
+	returnPointAndRangeKeys := func(eng storage.Engine) ([]storage.MVCCKeyValue, []storage.MVCCRangeKey) {
+		var rangeKeys []storage.MVCCRangeKey
+		var pointKeys []storage.MVCCKeyValue
+		for _, kvI := range storageutils.ScanKeySpan(t, eng, roachpb.Key("a"), roachpb.Key("z")) {
+			switch kv := kvI.(type) {
+			case storage.MVCCRangeKeyValue:
+				rangeKeys = append(rangeKeys, kv.RangeKey)
+
+			case storage.MVCCKeyValue:
+				pointKeys = append(pointKeys, kv)
+
+			default:
+				t.Fatalf("unknown type %t", kvI)
+			}
+		}
+		return pointKeys, rangeKeys
+	}
+
+	fingerprint := func(t *testing.T, startKey, endKey string, startTime, endTime hlc.Timestamp, allRevisions bool) int64 {
+		decimal := eval.TimestampToDecimal(endTime)
+		var fingerprint int64
+		query := fmt.Sprintf(`SELECT * FROM crdb_internal.fingerprint(ARRAY[$1::BYTES, $2::BYTES], $3, $4) AS OF SYSTEM TIME %s`, decimal.String())
+		require.NoError(t, sqlDB.QueryRow(query, roachpb.Key(startKey), roachpb.Key(endKey), startTime.GoTime(), allRevisions).Scan(&fingerprint))
+		return fingerprint
+	}
+
+	// Disable index recommendation so that >5 invocations of
+	// `crdb_internal.fingerprint` does not result in an additional call for
+	// generating index recommendations.
+	_, err := sqlDB.Exec(`SET CLUSTER SETTING sql.metrics.statement_details.index_recommendation_collection.enabled = false`)
+	require.NoError(t, err)
+
+	t.Run("fingerprint-empty-store", func(t *testing.T) {
+		fingerprint := fingerprint(t, "a", "z", hlc.Timestamp{WallTime: 0},
+			hlc.Timestamp{WallTime: timeutil.Now().UnixNano()}, true /* allRevisions */)
+		require.Zero(t, fingerprint)
+	})
+
+	s := serv.(*server.TestServer)
+	defer s.Stopper().Stop(ctx)
+
+	store, err := s.Stores().GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
+	eng := store.Engine()
+
+	// Insert some point keys.
+	txn := db.NewTxn(ctx, "test-point-keys")
+	pointKeysTS := hlc.Timestamp{WallTime: timeutil.Now().Round(time.Microsecond).UnixNano()}
+	require.NoError(t, txn.SetFixedTimestamp(ctx, pointKeysTS))
+	require.NoError(t, txn.Put(ctx, "a", "value"))
+	require.NoError(t, txn.Put(ctx, "b", "value"))
+	require.NoError(t, txn.Put(ctx, "c", "value"))
+	require.NoError(t, txn.Put(ctx, "d", "value"))
+	require.NoError(t, txn.Commit(ctx))
+
+	// Run a scan to force intent resolution.
+	_, err = db.Scan(ctx, "a", "z", 0)
+	require.NoError(t, err)
+
+	pointKeys, rangeKeys := returnPointAndRangeKeys(eng)
+	require.Len(t, pointKeys, 4)
+	require.Len(t, rangeKeys, 0)
+
+	// The store will have:
+	//
+	// ts2 [----------- rt -------------)
```

Comment:
Am I misreading? I don't think the store has a range tombstone until line 145 below?

Reply:
Nope my bad, fixed the comment.